### ISSUE TITLE
Drawer and element with visibility: hidden causes focus trapping to fail

### ISSUE THREAD

**Issue by tommhuth** [mantinedev/mantine#7036]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.13.4

### What package has an issue?

@mantine/core

### What framework do you use?

Next.js

### In which browsers you can reproduce the issue?

Chrome

### Describe the bug

I have a `Drawer` with a top link that was styled as `visibility: hidden` (it was a logo/home link, that visually made sense to not repeat on larger screens) and experienced some very strange focus trap issues. 

Focus would jump to my Drawer when opened, but when tabbing through and reaching the end of the modal, focus would stop at last focusable element and stay there (and not loop back to top). I could still tab backwards manually, but focus was not trapped to my modal and tabbing would escape out of my drawer. 

When I realized the cause, I figured the link is good to have accessible for everybody and so I solved it by setting `opacity: 0` and on focus set opacity to 1. But it made me wonder if me using `visibility: hidden` is bad practice or if I found a bug in Mantine's focus trapping? From what I can tell `visibility: hidden` is the same as `display: none`, except that it does not affect css layout and so I would perhaps expect focus trapping to ignore those focusable elements. 

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/mantine-react-template-forked-m4pspg

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
`visibility:hidden` styles cannot be processed correctly with use-focus-trap hook that Drawer uses for focus trap. The solution is to use `display:none` – it removes the element from the DOM and tab order. Currently, it is not planned to support `visibility:hidden` in use-focus-trap hook.
